numpy.intc not in _meshtags_types.keys() on 32-bit systems

Python unit tests show dolfinx.mesh.MeshTags failure on 32-bit architectures,
see https://ci.debian.net/packages/d/dolfinx/testing/i386/
https://ci.debian.net/packages/d/dolfinx/testing/armhf/
https://ci.debian.net/data/autopkgtest/testing/i386/d/dolfinx/9031796/log.gz
(search for `Datatype <class 'numpy.intc'> of values array not recognised` in the log)

One example of the error is:
```
_______________________ test_assembly_dx_domains[mode0] ________________________

mode = <GhostMode.none: 0>

    @pytest.mark.parametrize("mode", [dolfinx.cpp.mesh.GhostMode.none, dolfinx.cpp.mesh.GhostMode.shared_facet])
    def test_assembly_dx_domains(mode):
        mesh = dolfinx.generation.UnitSquareMesh(MPI.COMM_WORLD, 10, 10, ghost_mode=mode)
        V = dolfinx.FunctionSpace(mesh, ("CG", 1))
        u, v = ufl.TrialFunction(V), ufl.TestFunction(V)
    
        # Prepare a marking structures
        # indices cover all cells
        # values are [1, 2, 3, 3, ...]
        cell_map = mesh.topology.index_map(mesh.topology.dim)
        num_cells = cell_map.size_local + cell_map.num_ghosts
        indices = numpy.arange(0, num_cells)
        values = numpy.full(indices.shape, 3, dtype=numpy.intc)
        values[0] = 1
        values[1] = 2
>       marker = dolfinx.mesh.MeshTags(mesh, mesh.topology.dim, indices, values)

python/test/unit/fem/test_assemble_domains.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

mesh = <dolfinx.cpp.mesh.Mesh object at 0xeb0e7b68>, dim = 2
indices = array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,...179, 180, 181,
       182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194,
       195, 196, 197, 198, 199])
values = array([1, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
       3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,..., 3, 3, 3, 3, 3, 3,
       3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
       3, 3], dtype=int32)

    def MeshTags(mesh, dim, indices, values):
    
        if isinstance(values, int):
            values = numpy.full(indices.shape, values, dtype=numpy.int32)
        elif isinstance(values, float):
            values = numpy.full(indices.shape, values, dtype=numpy.double)
    
        dtype = values.dtype.type
        if dtype not in _meshtags_types.keys():
>           raise KeyError("Datatype {} of values array not recognised".format(dtype))
E           KeyError: "Datatype <class 'numpy.intc'> of values array not recognised"

/usr/lib/petsc/lib/python3/dist-packages/dolfinx/mesh.py:123: KeyError
```

mesh.py registers int8, int32, int64 and double from numpy in `_meshtags_types`, but not intc.  Should `numpy.intc` be explicitly added to the list?

I'm not sure if this is related to Issue #1253 or not.

--- a/python/test/unit/io/test_xdmf_meshtags.py
+++ b/python/test/unit/io/test_xdmf_meshtags.py
@@ -35,18 +35,18 @@ def test_3d(tempdir, cell_type, encoding):
     mesh = UnitCubeMesh(comm, 4, 4, 4, cell_type)
 
     bottom_facets = locate_entities(mesh, 2, lambda x: np.isclose(x[1], 0.0))
-    bottom_values = np.full(bottom_facets.shape, 1, dtype=np.intc)
+    bottom_values = np.full(bottom_facets.shape, 1, dtype=np.int32)
     left_facets = locate_entities(mesh, 2, lambda x: np.isclose(x[0], 0.0))
-    left_values = np.full(left_facets.shape, 2, dtype=np.intc)
+    left_values = np.full(left_facets.shape, 2, dtype=np.int32)
 
     indices, pos = np.unique(np.hstack((bottom_facets, left_facets)), return_index=True)
     mt = MeshTags(mesh, 2, indices, np.hstack((bottom_values, left_values))[pos])
     mt.name = "facets"
 
     top_lines = locate_entities(mesh, 1, lambda x: np.isclose(x[2], 1.0))
-    top_values = np.full(top_lines.shape, 3, dtype=np.intc)
+    top_values = np.full(top_lines.shape, 3, dtype=np.int32)
     right_lines = locate_entities(mesh, 1, lambda x: np.isclose(x[0], 1.0))
-    right_values = np.full(right_lines.shape, 4, dtype=np.intc)
+    right_values = np.full(right_lines.shape, 4, dtype=np.int32)
 
     indices, pos = np.unique(np.hstack((top_lines, right_lines)), return_index=True)
     mt_lines = MeshTags(mesh, 1, indices, np.hstack((top_values, right_values))[pos])

